internal/aws/proxy/conn.go

```diff
@@ -290,3 +306,15 @@ func getPartition(region string) string {
 	p, _ := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), region)
 	return p.ID()
 }
+
+func CreateIMDSV2AndFallbackClient(cfg awsSDKV2.Config) (*imds.Client, *imds.Client) {
```

Comment:
This is duplicated. Shouldn't there be a single common function for creating an IMDS client?

Reply:
The problem is that we are putting this in internal packages. Thus, I was not certain how to share between diff packages.